test/gtest/ucp/test_ucp_tag_offload.cc

```diff
@@ -281,3 +279,179 @@ UCS_TEST_P(test_ucp_tag_offload_multi, recv_from_multi)
 
 UCP_INSTANTIATE_TEST_CASE_TLS(test_ucp_tag_offload_multi, all_rcdc,
                               "\\rc,\\dc,\\ud,rc_x,dc_x")
+
+
+#if ENABLE_STATS
+
+class test_ucp_tag_offload_stats : public test_ucp_tag_offload_multi {
+public:
+
+    void init()
+    {
+        stats_activate();
+        test_ucp_tag_offload::init(); // No need for multi::init()
+    }
+
+    void cleanup()
+    {
+        test_ucp_tag_offload::cleanup();
+        stats_restore();
+    }
+
+    request* recv_nb_exp(void *buffer, size_t count, ucp_datatype_t dt,
+                         ucp_tag_t tag, ucp_tag_t tag_mask)
+    {
+        request *req1 = recv_nb_and_check(buffer, count, DATATYPE, tag,
+                                          UCP_TAG_MASK_FULL);
+
+        // Post and cancel another receive to make sure the first one was offloaded
+        request *req2 = recv_nb_and_check(buffer, count, DATATYPE, tag,
+                                          UCP_TAG_MASK_FULL);
+        req_cancel(receiver(), req2);
+
+        return req1;
+    }
+
+    ucs_stats_node_t* worker_offload_stats(entity &e)
+    {
+        return e.worker()->tm_offload_stats;
+    }
+
+    void validate_offload_counter(uint64_t rx_cntr, uint64_t val)
+    {
+        uint64_t cnt;
+        cnt = UCS_STATS_GET_COUNTER(worker_offload_stats(receiver()), rx_cntr);
+        EXPECT_EQ(val, cnt);
+    }
+
+    void wait_counter(ucs_stats_node_t *stats, uint64_t cntr,
```

Comment:
looks like this function is copied from base class, can we avoid the code duplication?